.github/workflows/tag-and-publish.yml

```diff
@@ -4,9 +4,16 @@ name: Tag and publish
 # For future functionality like 'publish to Artifactory', 'push to GCR', ..., see the data catalog version of this file.
 
 on:
-  push:
-    branches: [ main ]
-    paths-ignore: [ '*.md' ]
+  workflow_call:
+    secrets:
+      BROADBOT_TOKEN:
+        required: true
+      ARTIFACTORY_USERNAME:
+        required: true
+      ARTIFACTORY_PASSWORD:
+        required: true
+      DSDEJENKINS_DOCKERHUB_PASSWORD:
+        required: true
```

Comment:
Out of interest, do you know why these are required now when they weren't needed before?

Reply:
It's because the `tag-and-publish` workflow is now being triggered as a `workflow_call` by the parent workflow  `build-and-test`. The `workflow_call` needs these parameters to be passed from the parent workflow.